src/value/number.rs

```diff
@@ -22,6 +48,14 @@ pub enum Number {
     U128(u128),
     F32(F32),
     F64(F64),
+    #[cfg(not(doc))]
+    #[allow(private_interfaces)]
+    __NonExhaustive(private::Never),
```

Comment:
Unsure whats going on but this didn't make the enum exhaustive.  I wonder if the compiler recognized that this variant cannot be constructed so therefore it doesn't need to be matched?

Reply:
Uff, maybe this comes from some of the newer uninhabited type matching ...

Since `private::Never` is private, we can make the type inhabited but not constructible. Would changing this be a breaking change?

Reply:
Currently, people can exhaustively match on `Number`.  Making it actually non-exhaustive would then be a breaking change.

Reply:
We’ll definitely need a compile-fail test then to ensure there are no more slip ups - I’m sorry that the first fix didn’t catch this.